src/OpenSage.Game/Logic/OrderGeneratorSystem.cs

```diff
@@ -176,7 +177,7 @@ public void StartConstructBuilding(ObjectDefinition buildingDefinition)
 
         public void SetRallyPoint()
         {
-            ActiveGenerator = new RallyPointOrderGenerator(Game);
+            ActiveGenerator = new RallyPointOrderGenerator(Game, Game.PlayerManager.LocalPlayer.SelectedUnits.First());
```

Comment:
what happens if multiple structures are selected? I can't remember if that's even possible...

Reply:
It's not possible to select more than one structure. But I guess it would be nice to add some validation to clarify the intention there.

Reply:
we _could_ also use `Single()`.

There is also the case of slaved items (like when you select a stinger site). Interestingly, in Generals, if you select a tunnel network and order it to attack, it orders the rpg troops created from the tunnel network to attack as well! But I don't know if this is actually a multiple selection, or just orders given to slaved units